classes/04class/exercises/c04-iac02/nimmi89/vpc.tf

```diff
@@ -0,0 +1,104 @@
+# Create VPC/Subnet/Security Group/Network ACL
+
+#VPC
+resource "aws_vpc" "tf-vpc" {
+  cidr_block = var.vpcCIDRblock
+  tags = {
+    Name = "devopsacademy-i"
+  }
+}
+
+
+#Public subnets -1 and 2
+resource "aws_subnet" "public" {
+  count = length(var.public_subnet_cidr_blocks)
```

Comment:
the way `count` is used here, it is expecting ordering in the input variables which is not clear in the code. Example: if the `name` and `cidr` are swapped in the input the result would be wrong.

better to refactor the input variables and use `map` and `objects`, so `for_each` could be used here.